Fix UFC coordinate input for preplanned targets on F/A-18

mzltn opened the pull request:
Changes:
 - Add methods to PrePlannedCoordinate to extract values from coordinate: Convert_to_DegMin, Convert_to_DegMinSec.
 - Refactor BuildCoordinate method (namespace DTC.Models.FA18.Upload, class PrePlannedBuilder).

p-louis committed:
Added usage-hint on sequence-page

p-louis committed:
Adapted Load/Save Page to hornet

p-louis committed:
Improvements for PrePlanned features

ADDED Overview page to view all the currently set coordinates in single screen
ADDED support for J-109 aka GBU31-3/4

p-louis committed:
Fixed bug where deselcting Sequences would actually deselect PP

p-louis committed:
Fixed some issues

- CMS Reference programs 1 and 5 held wrong values
- COMM2 frequency entry sufferred from lack of delay after pressing ENT
- Checking Sequences on upload page would enter MFD page even though no sequences are selected
- Increased some delays to ensure stability of data entry.

p-louis committed:
Corrected Typo in Readme

p-louis committed:
FIXED: Loading of pre-planned data from file/clipboard

p-louis committed:
Merge branch 'feature/fa18' of https://github.com/p-louis/dcs-dtc into feature/fa18

p-louis committed:
Increased Wait times from coordinate entry

p-louis committed:
FIXED issue with matching CMS intervals

p-louis committed:
Added Pipeline for automatic artifact upload to release

p-louis committed:
CHANGED Workflow name for Release Workflow

p-louis committed:
FIX artifact path for release pipeline

p-louis committed:
CHANGE release pipeline no longer triggered by drafts

p-louis committed:
ADDED Version number Display
FIXED Issue where F/A-18 waypoint system would get confused

p-louis committed:
Added check for maximum waypoint altitude in waypoint-edit dialog

mzltn committed:
Replace numbered variables with Dictionary objects

replace:
 - PrePlannedCoordinate PP1, PP2, PP3, PP4, PP5;
    with
   Dictionary<int, PrePlannedCoordinate> PP;

 - PrePlannedStation Sta2, Sta3, Sta7, Sta8;
    with
   Dictionary<int, PrePlannedStation> Stations;

mzltn committed:
Refactor PrePlannedPage.cs

changes:
  - group controls of PrePlannedPage.cs into a per-station UserControl: PrePlannedStationControl.cs
  - use a single event handler for PP1,2,3,4,5 checkboxes
  - use a single event handler for PP1,2,3,4,5 buttons

mzltn committed:
Refactor F/A-18 PrePlannedBuilder class

The stations are grouped by payload type by GroupStationsByPayloadType
into a List<List<PrePlannedStation>>.

The outer List contains the groups, which are in the order by which the
payload type was first encountered while iterating through stations in
order 2,3,7,8 (This matches the order as payload types are listed on the
MFD).

The inner List contains the individual stations of that group, which are
ordered as 8,2,7,3 (stepping order of stations with the same payload).

Other changes:
 - Changed InputStation to only press PP1 once at the end to return to
   it if needed.
 - Renamed isJdam to isJdamOrJsow
 - Simplified switch-case structure in GetDsplyCommand and isJdamOrJsow
   for readability.

mzltn committed:
Catch exception if a connection error occurs when trying to upload

Gosling616 commented:
> Changes:
> 
>  - Add methods to PrePlannedCoordinate to extract values from coordinate: Convert_to_DegMin, Convert_to_DegMinSec.
> 
>  - Refactor BuildCoordinate method (namespace DTC.Models.FA18.Upload, class PrePlannedBuilder).

Gosling616 commented:
The Preplanned coordinates are correct in that the numbers are the correct numbers to input into the JDAMs. These numbers are already in DD MM SS.ss however, they are displayed incorrectly as DD MM.mmmm.  But they are NOT Decimal Minute numbers. They are Decimal Second numbers already.

mzltn commented:
Thanks for the review. I have somehow overlooked this case. I have checked, and the coordinate capture feature also gives the same format that you mentioned.
In this case however, I think the TextBox mask should be set to follow the same format, as the current one is somewhat misleading.

mzltn committed:
Fix UFC coordinate input for preplanned targets on F/A-18

Changes:
  - change mask of DTCTextBox of PrePlannedEdit
  - change validating regex of PrePlannedCoordinate
  - Refactor BuildCoordinate method (namespace DTC.Models.FA18.Upload, class PrePlannedBuilder).

mzltn committed:
Add support for SLAM-ER Steerpoints on F/A-18

mzltn committed:
Change MFDs to a known state before upload

p-louis commented on dcs-dtc/Models/FA18/PrePlanned/PrePlannedStation.cs:
```diff
@@ -10,18 +10,17 @@ public class PrePlannedStation
-        public PrePlannedCoordinate PP3 {  get; set; }  
-        public PrePlannedCoordinate PP4 {  get; set; }  
-        public PrePlannedCoordinate PP5 {  get; set; }  
+        public int stationNumber { get; set; }
```
stationNumber should not have a setter as it should be fixed once initialized

p-louis reviewed (approved):
Hey! Thanks for cleaning up the mess I made with the PP stuff 😃 

And sorry I didn't react sooner, I usually don't check for new PRs on this repo because most people open theirs on the upstream one.

Overall LGTM
